src/ArchiveBuilderHelper.cpp

```diff
@@ -1,66 +1,75 @@
 #include <bsapacker/ArchiveBuilderHelper.h>
 
-#include <QDirIterator>
-#include <QtConcurrent/QtConcurrentMap>
+#include <algorithm>
+#include <boost/algorithm/string.hpp>
 
 #include "SettingsService.h"
 
+using std::filesystem::path;
+using std::filesystem::directory_entry;
+using std::filesystem::directory_iterator;
+using std::filesystem::recursive_directory_iterator;
+
 namespace BsaPacker
 {
-	const std::array <QString, 3> ArchiveBuilderHelper::INCOMPRESSIBLE_TYPES = { "wav", "ogg", "mp3" };
+	const std::array <std::string, 3> ArchiveBuilderHelper::INCOMPRESSIBLE_TYPES = { ".wav", ".ogg", ".mp3" };
 
 	ArchiveBuilderHelper::ArchiveBuilderHelper(const ISettingsService* settingsService)
 		: m_SettingsService(settingsService)
 	{
 	}
-	uint32_t ArchiveBuilderHelper::getFileCount(const QDir& rootDirectory) const
+	uint32_t ArchiveBuilderHelper::getFileCount(const path& rootDirectory) const
 	{
-		QDirIterator it(rootDirectory, QDirIterator::Subdirectories);
 		uint32_t count = 0;
-		while (it.hasNext()) {
-			it.next();
+		for(auto& p : recursive_directory_iterator(rootDirectory)) {
 			count++;
 		}
 		return count;
 	}
 
-	QStringList ArchiveBuilderHelper::getRootDirectoryFilenames(const QDir& rootDirectory) const
+	std::vector<std::string> ArchiveBuilderHelper::getRootDirectoryFilenames(const path& rootDirectory) const
 	{
-		const std::function<QString(const QFileInfo&)> getFilename = [](const QFileInfo& fi) {
-			return fi.fileName();
-		};
-		return QtConcurrent::blockingMapped(rootDirectory.entryInfoList(QDir::Files),
-											getFilename);
+		std::vector<std::string> filenames;
+		for (const auto& entry : directory_iterator(rootDirectory)) {
+			filenames.push_back(entry.path().filename().string());
```

Comment:
You should probably use `native()` and replace `std::string` with `path::string_type` (which is `wstring` on Windows). The behavior of `string()` with `string_type` = `wstring` is unspecified by the standard.

Reply:
I've changed it to use `path::string_type`